src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -1,6 +1,218 @@
 using LinearAlgebra: checksquare
 using LinearAlgebra.BLAS: gemv, gemv!, gemm!, trsm!, axpy!, ger!
 
+#####
+##### `lu`
+#####
+
+# These rules are necessary because the primals call LAPACK functions
+
+# frule for square matrix was introduced in Eq. 3.6 of
+# de Hoog, F.R., Anderssen, R.S. and Lukas, M.A. (2011)
+# Differentiation of matrix functionals using triangular factorization.
+# Mathematics of Computation, 80 (275). p. 1585.
+# doi: http://doi.org/10.1090/S0025-5718-2011-02451-8
+# for derivations for wide and tall matrices, see
+# https://sethaxen.com/blog/2021/02/differentiating-the-lu-decomposition/
```

Comment:
This is great!

Can we add it to the `More Examples` section of the `Deriving Array Rules` docs page?

E.g. some like
```md
For Differentiating the LU decomposition, see [this blog post by Seth Axen](https://sethaxen.com/blog/2021/02/differentiating-the-lu-decomposition/).
```

Reply:
This is a good idea. I currently have an open PR to update that docs page, and I'll include this there: https://github.com/JuliaDiff/ChainRulesCore.jl/pull/290